src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -28,6 +29,35 @@ internal static class Constants
     /// </summary>
     public class ModuleIntrinsics
     {
+#if !UNIX
+        /// <summary>
+        /// Compares a PATH entry on Windows. Treats / and \ as the same and
+        /// compares using OrdinalIgnoreCase.
+        /// </summary>
+        private class WindowsPathEnvComparer : IEqualityComparer<char>
+        {
+            public bool Equals(char x, char y)
+            {
+                if (x == y)
+                {
+                    return true;
+                }
+                // x is always canonicalised so we only need to compare if y is /.
+                else if (x == Path.DirectorySeparatorChar && y == Path.AltDirectorySeparatorChar)
+                {
+                    return true;
+                }
+
+                return MemoryMarshal.CreateReadOnlySpan(ref x, 1).CompareTo(
```

Comment:
Yes, the various Apple platforms use case-_insensitive_ file-systems by default (though at least on macOS you could by opt-in use a case-sensitive one instead, so checking by platform is, strictly speaking, not enough). However, `\` and `/` should _not_ be considered interchangeable there (which isn't a problem here, given that both `Path.DirectorySeparatorChar` and `Path.AltDirectorySeparatorChar` report `/` on Unix-like platforms).

Reply:
From my perspective because these platforms can be case sensitive we should err on the side of caution by checking using a case sensitive comparison on them. Even if the FS is case insensitive the worst that happens is that there is a duplicate in the path which shouldn’t affect anything except the extra memory to store the env var.

Reply:
(Re `\` and `/`: yes, as I've stated, the code would work as-is; my concern was that the Windows-only comments may be confusing should the code also be used for Apple platforms.)
I agree with your assessment that it's probably fine to keep the case-sensitive behavior on Apple platforms (saves a method call, avoids edge cases).

Reply:
> I agree with your assessment that it's probably fine to keep the case-sensitive behavior on Apple platforms (saves a method call, avoids edge cases).

Me too. @jborean93 Please add a comment about the conclusion.

Reply:
Done

Reply:
Please add a comment based on "Even if the FS is case insensitive the worst that happens is that there is a duplicate in the path which shouldn’t affect anything except the extra memory to store the env var." for MacOS and IOS

Reply:
Is https://github.com/PowerShell/PowerShell/pull/24568/files#diff-127b1a58979f0695aff5b35c0407bb15db9596865840c1f2d58161856b5e1ae2R1174-R1177 not enough? Do we even create a build for iOS anyway?

Reply:
I see. Thanks!